### PR TITLE
Draft: Start adding SPDX-License-Identifier comments

### PR DESCRIPTION
**Pull request by gotmax23**
Automated with:

```
mkdir LICENSES
ln -sr COPYING LICENSES/GPL-3.0-or-later.txt
git mv PSF-license.txt LICENSES/PSF-2.0.txt
reuse addheader -l PSF-2.0 plugins/module_utils/_version.py
rg -l 'GNU General Public License v3.0+' | parallel reuse addheader -l GPL-3.0-or-later
```

Signed-off-by: Maxwell G <gotmax@e.email>

##### SUMMARY
This is a WIP PR to show what adopting REUSE would look like. I know that this still needs a changelog fragment. I will add that once we decide on this. I have included how I've automated this so folks can see what the process looks like.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Licensing

##### ADDITIONAL INFORMATION
This is currently being discussed in https://github.com/ansible-community/community-topics/issues/112.

### CODE REVIEW

**Comment by gotmax23** [comment]
Current output of `reuse lint`: https://gist.github.com/gotmax23/b82d00caf4602d95fd0e379a1c1ac47c

I only added license comments for plugins/module_utils/_version.py and the files containing `GNU General Public License v3.0+`, so there are a lot of files missing license information. I can add bulk add `GPL-3.0-or-later` license headers to other files that don't currently have licensing information if you confirm that there's no other secondary licenses I missed.

File: plugins/connection/docker.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 # (c) 2015, Leendert Brouwer (https://github.com/objectified)
 # (c) 2015, Toshio Kuratomi <tkuratomi@ansible.com>
 # Copyright (c) 2017 Ansible Project
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```
Comment (felixfontein) [inline plugins/connection/docker.py:7]: This needs to stay, otherwise it's impossible to pass the sanity tests.

Comment (gotmax23) [reply plugins/connection/docker.py:7]: So, it turns out that `reuse addheader` is not supposed to remove the old license headers; it's a bug. Currently, it completely replaces each "comment block" that contains any copyright/license information with its own formatted version. See https://github.com/fsfe/reuse-tool/issues/338. I came up with an absolutely terrifying incantation to separate `# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)` with a newline so it doesn't get clobbered.

**Comment by felixfontein** [comment]
I created #430 with https://gist.github.com/felixfontein/3d9502e22d1461dc3baadddeeb9d9cd9 and some manual work.

**Comment by gotmax23** [comment]
Closing in favor of #430

**Pull request closed without merging**